resource/Resource.md

```diff
@@ -0,0 +1,154 @@
+# Resource API Overview
+The resource library primarily defines a type that captures information about the entity
+for which stats or traces are recorded. It further provides a framework for detection of
+resource information from the environment and progressive population as signals propagate
+from the core instrumentation library to a backend's exporter.
+
+## Resource type
+A `Resource` describes the entity for which a signal was collected through two fields:
+* `type`: a string which describes a well-known type of entity. It SHOULD be namespaced
+to avoid collisions across different environment, e.g. `k8s.io/container`,
+`cloud.google.com/gce/instance`. 
+* `labels`: a dictionary of labels with string keys and values that provide information
+about the entity.
+
+Type strings and label keys MAY start with a domain name and MAY be further namespaced through
+single slashes. All other characters MUST be alphanumeric. Label values MAY contain any valid
+unicode character.
```

Comment:
Right, that should've just said "utf8" probably. But I'm fine with a subset, which it will be anyway in practice. 
Only thing I'd be worried about is, that that might not add up with backends and thus make OC restrictive without much gain.

Prometheus supports utf8 strings for label values and it hasn't been an issue yet.

Reply:
We (ab)use prometheus at work and have much wider range in use than ASCII. 👍 to UTF-8.